pkg/stub/status_test.go

```diff
@@ -0,0 +1,393 @@
+package stub
+
+import (
+	"testing"
+
+	ingressv1alpha1 "github.com/openshift/cluster-ingress-operator/pkg/apis/ingress/v1alpha1"
+	"github.com/openshift/cluster-ingress-operator/pkg/util/clusteroperator"
+	osv1 "github.com/openshift/cluster-version-operator/pkg/apis/operatorstatus.openshift.io/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestSetStatusConditions(t *testing.T) {
```

Comment:
1. Given inputs, the output should be some expected `[]ClusterOperatorStatusCondition`. Existing condition isn't an input to the status computation function. When would `old` ever be relevant?
2. What about doing a few tweaks to allow expressing the cases as one-liners for easier comprehension?

Reply:
> Existing condition isn't an input to the status computation function. When would old ever be relevant?

It is, and I think it needs to be if for no other reason so that we update `LastTransitionTime` if and only if it appropriate to do so (although the unit tests ignore `LastTransitionTime` presently).  As far as testing, having the old conditions lets us test to make sure `SetStatusCondition` doesn't add duplicate or fail to update an existing condition, for example. 

> What about doing a few tweaks to allow expressing the cases as one-liners for easier comprehension?

Maybe something like this?
```go
type testInputs struct {
  numWanted, numAvailable, numMissing, numUnavailable int
}
type testOutputs struct {
  failing, progressing, available bool
}
// ...
  testCases := []struct {
    description string
    haveNamespace bool
    in testInputs
    out testOutputs
  }{
    { "1/2 ingresses available", testInputs{2, 1, 0, 1}, testOutputs{false, true, false} },
    // ...
```
and ignore condition reason and message? (That example is >80 characters but technically a one-liner...)

Reply:
>>Existing condition isn't an input to the status computation function. When would old ever be relevant?

>It is, and I think it needs to be if for no other reason so that we update LastTransitionTime if and only if it appropriate to do so (although the unit tests ignore LastTransitionTime presently). 

I guess it could make sense in this case. I'm not sure how important that is.

>As far as testing, having the old conditions lets us test to make sure SetStatusCondition doesn't add duplicate or fail to update an existing condition, for example.

Assuming our operator owns that field, whenever the operator wants to update the field it could atomically replace the entire field with newly computed contents. Adding complexity to diff and patch seems like overkill.

Reply:
I implemented something very similar to what I outlined in my last comment.

Reply:
Just saw your last two comments.  I like using additional types such as your `daemonset` type for clarity, but I don't think it matters in the test case how we name the daemonsets or what specific value (besides zero or non-zero) we use for `NumberAvailable`.